framework/material/Icon.kt

```diff
@@ -43,7 +43,83 @@ object Icon {
     data class Offset(val x: Dp, val y: Dp)
     data class Scale(val x: Float, val y: Float)
 
-    enum class Code(
+    enum class Purpose(val shape: Shape) {
```

Comment:
Purpose is definitely a more ergonomic thing to group icons by than shape. However, I don't think `Purpose` is an appropriate name for such a construct.

This is the enum that we use values from to define the icons in use across the application. Passing an `Icon.Code` as a parameter named `icon` was... fine: clearly the code represents the icon, but to be completely precise, the parameter should arguably have been named `iconCode`.

If we were to follow the same logic now, we'd end up naming our parameters `iconPurpose`, which simply doesn't mean what we want it to. Moreover, the name `icon` is frankly the best parameter name for representing an icon. So how about this: we move all of these enum members up one level, by refactoring `Icon` itself to be an enum? Everything else should stay the same - for example, `Icon.Shape`.

(By the way, another thing: I note we use SCREAMING_SNAKE_CASE for the enum members, which is a Java convention. I believe the Kotlin convention is to use PascalCase for enum members. Can we check the rest of Studio to see if there is precedent for one or the other?)

Reply:
I did not know you could have an enum class containing objects and etc. TIL! 

Regarding the case used for enums, unfortunately we're a little split internally. See the following examples:
`Geometry.kt`
```
    enum class AngularDirection {
        Clockwise,
        CounterClockwise
    }
```
`KeyMapper.kt`
```
enum class Command {
        MOVE_CHAR_LEFT,
        MOVE_CHAR_RIGHT,
        MOVE_WORD_RIGHT,
        MOVE_WORD_LEFT,
        MOVE_PARAGRAPH_PREV,
        MOVE_PARAGRAPH_NEXT,
...
```

There's only a few instances of PascalCase, coincidentally all written by you! 

Given our discussion in person and in line with Kotlin's convention (https://kotlinlang.org/docs/enum-classes.html) of using SCREAMING_SNAKE_CASE for enum class members, I've refactored out all the uses of PascalCase.